src/core/tsi/s2a/handshaker/s2a_handshaker_client.h

```diff
@@ -0,0 +1,122 @@
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef GRPC_CORE_TSI_S2A_HANDSHAKER_S2A_HANDSHAKER_CLIENT_H
+#define GRPC_CORE_TSI_S2A_HANDSHAKER_S2A_HANDSHAKER_CLIENT_H
+
+#include <grpc/byte_buffer.h>
+#include <grpc/byte_buffer_reader.h>
+#include <grpc/grpc.h>
+#include "src/core/lib/iomgr/closure.h"
+#include "src/core/lib/iomgr/pollset_set.h"
+#include "src/core/tsi/transport_security_interface.h"
+
+namespace grpc {
+namespace experimental {
+
+typedef struct s2a_tsi_handshaker s2a_tsi_handshaker;
+
+/** The Secure Session Agent (S2A) handshaker client interface. It facilitates
+ *  establishing a secure channel with the peer by interacting with the
+ *  S2A's handshaker service. More precisely, it schedules
+ *  a handshaker request that could be one of client_start, server_start,
+ *  and next handshaker requests. The interface and all API's are
+ *  thread-compatible. **/
+typedef struct s2a_handshaker_client s2a_handshaker_client;
+
+typedef struct grpc_s2a_credentials_options grpc_s2a_credentials_options;
```

Comment:
It seems more appropriate to define it at a higher level in the security stack (e.g., credential)?

Reply:
It will be; this is just a placeholder so that this PR compiles. (For reference, it is declared and defined in this PR: https://github.com/matthewstevenson88/grpc/pull/2 .)